.github/workflows/nightly.yml

```diff
@@ -0,0 +1,73 @@
+name: Nightly build
+
+on:
+  schedule:
+    - cron: "0 3 * * *"
+  push:
+    branches:
+      - master
+      - main
+
+jobs:
+  build:
+    name: Build integration for
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        goos: [ linux ]
+        goarch: [ amd64, arm64, arm ]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
```

Comment:
Could we also cache the intermediate build objects? Just adding the path like we do here should suffice: https://github.com/actions/cache/blob/main/examples.md#linux

Reply:
Sure, though I'll do it separately, as this is copy-paste from the release pipeline.

Reply:
Done.